kong/tools/public.lua

```diff
@@ -175,10 +175,6 @@ do
         return node_id
       end
 
-      if ngx.get_phase() == "init" then
```

Comment:
Was this not needed anymore? If not wanted, what was the reason?

Reply:
This was a safeguard against using the PRNG (to generate the node id) in the master process (since the PRNG was not seeded yet). Since we now seed the PRNG in the master process, this safeguard can be removed. Service Mesh needed to use the node id in the init context, hence the changes.